src/test/RippleKey_test.cpp

```diff
@@ -0,0 +1,358 @@
+                    "F1E26FEEF9819EC8E").first;
+
+            // Remove the signature
+            tx.makeFieldAbsent(sfTxnSignature);
+            tx.setAccountID(sfAccount, calcAccountID(key.publicKey()));
+            check = tx.checkSign(true);
+            BEAST_EXPECT(!check.first);
+            BEAST_EXPECT(check.second == "Invalid signature.");
+
+            // Now re-sign it
+            key.singleSign(tx);
+            BEAST_EXPECT(tx.checkSign(true).first);
+            // Same signature
+            BEAST_EXPECT(tx.getFieldVL(sfSigningPubKey) == expectedSigningKey);
+            BEAST_EXPECT(tx.getFieldVL(sfTxnSignature) == expectedSignature);
+            BEAST_EXPECT(!tx.isFieldPresent(sfSigners));
+        }
+
+        {
+            // Use a different key to multisign, because an account can't
+            // multisign its own transaction.
+            auto const key = RippleKey::make_RippleKey(kt, std::string("bob"));
+
+            // Now multisign it with the test key
+            key.multiSign(tx);
+            BEAST_EXPECT(tx.checkSign(true).first);
+            // No single signature
+            BEAST_EXPECT(!tx.isFieldPresent(sfTxnSignature));
+            BEAST_EXPECT(tx.getFieldVL(sfSigningPubKey).empty());
+            if (BEAST_EXPECT(tx.isFieldPresent(sfSigners)))
+            {
+                auto const& signers = tx.getFieldArray(sfSigners);
+                BEAST_EXPECT(signers.size() == 1);
+                auto const& signer = signers[0];
+                auto const expectedAccount = kt == KeyType::secp256k1 ?
+                    "rPMh7Pi9ct699iZUTWaytJUoHcJ7cgyziK" :
+                    "rJy554HmWFFJQGnRfZuoo8nV97XSMq77h7";
+                auto const expectedPubKey = kt == KeyType::secp256k1 ?
+                    "02691AC5AE1C4C333AE5DF8A93BDC495F0EEBFC6DB0DA7EB6"
+                    "EF808F3AFC006E3FE" :
+                    "ED3CC3D14FD80C213BC92A98AFE13A405A030F845EDCFD5E3"
+                    "95286A6E9E62BA638";
+                auto const expectedSignature = kt == KeyType::secp256k1 ?
+                    "304402200719B97DA805D72C51100ECFEA86F73B7AC787559E"
+                    "1AB34285C82CD0C7EC0A1402206EDDE8077DB49F808ED1BFC6"
+                    "6CC06B944A11F05B58D59247B027B40F04E95412" :
+                    "D12E9335B9AADAB917E65F5E3DB4B8A37DB0F5F5DC2E7333FF"
+                    "26A8E5FEEC203D1F65ACADE6E6D0BD8E01D21C1838DF005E66"
+                    "9AC1C8E57CA41405374CEDBB2309";
+                BEAST_EXPECT(to_string(signer.getAccountID(sfAccount)) ==
+                    expectedAccount);
+                BEAST_EXPECT(strHex(signer.getFieldVL(sfSigningPubKey)) ==
+                    expectedPubKey);
+                BEAST_EXPECT(strHex(signer.getFieldVL(sfTxnSignature)) ==
+                    expectedSignature);
+            }
+
+            // Sign with a second key
+            auto const key2 = RippleKey::make_RippleKey(kt,
+                passphrase);
```

Comment:
Thanks for checking those! Within the unit test, the `checkSign` is sufficient, but I appreciate the extra verification step.

As for `rpcMASTER_DISABLED`, I only see https://github.com/ripple/rippled/blob/develop/src/ripple/rpc/impl/TransactionSign.cpp#L131, which is checking the `lsfDisableMaster` flag for the particular account.